Isosurfacing_3/doc/Isosurfacing_3/Isosurfacing_3.txt

```diff
@@ -0,0 +1,218 @@
+MC runs over a 3D grid, i.e., a 3D domain partitioned into hexahedral cells, and processes all cells of the input 3D domain individually while sampling the input scalar field at the grid corners.
+Each cell corner is assigned a sign (+/-) to indicate whether its scalar field value is above or below the user-defined isovalue.
+MC creates a vertex for each grid edge with a sign change, i.e., where the edge intersects the isosurface.
+More specifically, the vertex position is computed via linear interpolation of the scalar field values evaluated at the cell corners forming the edge. Depending on the configuration of signs at the cell corners, the resulting vertices are connected to form triangles within the cell. The figure below illustrates the configurations in 2D. There is no less than 33 cases in 3D (not shown).
+
+
+\cgalFigureAnchor{isosurfacing_mc_cases}
+<center>
+<img src="mc_cases.png" style="max-width:70%;"/>
+</center>
+\cgalFigureCaptionBegin{isosurfacing_mc_cases}
+Different configurations of Marching Cubes in 2D.
+\cgalFigureCaptionEnd
+
+
+The proposed implementation is generic in that it can process any grid-based data structure that consists of hexahedral cells.
+In case of a conforming grid, MC generates as output a surface triangle mesh that is 2-manifold in most scenarios.
+% PA: reformulate, as it is either strict or not - precise in which cases it is not, and say  whether you talk about just combinatorially 2-manifold, or truly 2-manifold with 3D embedding taken into account
+If the mesh is 2-manifold and the isosurface does not intersect the domain boundary, then the output mesh is watertight. Conversely, the output mesh has boundaries when the isosurface intersects the domain boundaries (see Figure below).
+
+\cgalFigureAnchor{isosurfacing_mc_open}
+<center>
+<img src="open.png" style="max-width:70%;"/>
+</center>
+\cgalFigureCaptionBegin{isosurfacing_mc_open}
+The output MC mesh has boundaries when the isosurface intersects the domain boundary.
+\cgalFigureCaptionEnd
+
+
+
+
+As MC only proceeds by linear interpolation of the sampled scalar field along the grid edges,
+it can miss details or components that are not captured by the said sampling and interpolation.
+
+Compared to other approaches such as Delaunay refinement, and depending on the parameters,
+the MC algorithm often generates more triangles, and more skinny triangles with small or large angles.
+[The statement is too strong - I would largely reformulate it as this depends too much on the parameters.
+I recommend to add instead a figure with different parameters and methods,
+depicting the mesh edges in black in addition to the shaded facets]
+MC does not preserve the sharp features present in the isovalue of the input scalar field
+% PA: add a figure
+
+
+\subsection subsectmc Topologically Correct Marching Cubes (TMC)
+This algorithm is an extension to MC and provides additional guarantees for the output \cgalCite{g-ctcmi-16}.
```

Comment:
doxygen does not know this publication.  Can you fix it or give me the bibtex entry.